tests/github-glue.test.ts

```diff
@@ -0,0 +1,204 @@
+import { Octokit } from "@octokit/rest";
+import "jest";
+import { git, gitConfig } from "../lib/git";
+import { GitHubGlue } from "../lib/github-glue";
+
+/*
+This test requires setup.  It will run successfully if setup has
+not been done.  If the test fails, there may be a pull request and a
+branch to be deleted on github and the local repo.  The test will
+attempt cleanup of a previously failed test.
+
+Setup:
+gitgitgadget.githubtest.githubuser must be configured for this test to
+identify your GitHub login.
+The value must also be configured to identify a test repo to use as
+gitgitgadget.<login>.githubrepo.
+
+Additionally, a GitHub personal access token must be set for the
+login to succeed.  This can be restricted to the test repo.
+
+For example, these configuration settings are needed (where
+`octokity` is your GitHub login):
+git config --add gitgitgadget.githubtest.githubuser octokity
+git config --add gitgitgadget.octokity.githubrepo ggg-test
+git config --add gitgitgadget.octokity.githubtoken feedbeef...
+
+The test repo must exist in github and locally.  It is expected to be
+located at the same directory level as this project.
+*/
+
+class GitHubProxy extends GitHubGlue {
+    public octo = new Octokit();
+    public constructor(workDir?: string, repo = "git") {
+        super(workDir, repo);
+        this.octo = this.client;
+    }
+
+    public async authenticate(repositoryOwner: string): Promise<void> {
+        await this.ensureAuthenticated(repositoryOwner);
+        this.octo = this.client;
+    }
+}
+
+jest.setTimeout(180000);
+
+test("identify user", async () => {
+    const owner = await gitConfig(`gitgitgadget.githubtest.githubuser`) || "";
+    const repo = await gitConfig(`gitgitgadget.${owner}.githubrepo`) || "";
+
+    if (owner && repo) {
+        const userName = await gitConfig(`user.name`, `../${repo}`) || "";
+
+        const github = new GitHubProxy(`../${repo}`, repo);
```

Comment:
So the repository needs to be checked out next to the current worktree? If that is the case, I think we might want to add that to the description of the setup.

Reply:
Please suggest better wording for lines ~27-28.  When you are nose deep in something, what you think is obvious may not be.  I am open to (and appreciate) suggestions.

Reply:
Hmm. I fail to find a good wording. "Sibling directory"? "must live in the same directory as the current worktree"? I'm happy with neither.

Reply:
Adding `(ie ../)` to help clarify.